modules/doc/content/application_usage/failed_solves.md

```diff
@@ -67,44 +165,196 @@ Initial residual before setting preset BCs: 65444.1
-     14 Linear |R| = 5.510719e+04
-     15 Linear |R| = 5.510718e+04
+... (until l_max_its=15)
+     15 Linear |R| = 5.381315e+04
  Solve Did NOT Converge!
 ```
 
-In this case you are likely to require many nonlinear iterations as well, but the reason is that your linear iterations don't drop. This could be due to missing terms or errors in your Jacobian or because the way you are applying your preconditioner in PETSc is not good for the problem. Make sure your Jacobian is correct and add off-diagonal terms for multivariable problems.
+In this case you are likely to require many nonlinear iterations as well, but the reason is that your linear iterations don't drop. This could be due to missing terms or errors in your Jacobian -- especially for Newton's method, but for PJFNK the Jacobian is often used for preconditioning still -- or because the way you are applying your preconditioner in PETSc is not good for the problem. First, let's make sure your Jacobian is correct and has off-diagonal terms for multivariable problems.
+
+!alert note
+Additional resources on linear solve issues may be found in the
+[PETSc manual FAQ](https://petsc.org/release/faq)
+
+### Incorrect Jacobian
+
+First, it is important to understand where the Jacobian may come into play because this sets the bar for how accurate the
+Jacobian needs to be.
+For Newton's method, the linear solve is essentially the inversion of the Jacobian (an explicit inverse is generally **NOT** formed)
+and its action on the residual vector. As such, we generally consider that it is **essential** that the Jacobian be accurate.
+A solve with a poor Jacobian is unlikely to converge.
+For PJFNK, the linear systems are formed by computing the action of the Jacobian rather than the Jacobian. But the preconditioning
+often relies on forming the Jacobian. By default, only the diagonal of the Jacobian is computed for preconditioning. To form
+the entire Jacobian, including the variable coupling terms on the off-diagonals, this syntax may be used:
+
+!listing test/tests/preconditioners/auto_smp/ad_coupled_convection.i block=Preconditioning
+
+A basic check on whether the Jacobian may be inaccurate is to simply replace it with a Jacobian formed using finite differencing,
+using the [Finite Difference Preconditioner](FiniteDifferencePreconditioner.md). This replaces every call to `::computeJacobian` in the kernels and boundary
+conditions with finite differencing on the computation of the residual. If a case of poor convergence is fixed by this
+simple action, then the Jacobian was not good and it needs to be fixed.
+
+!listing test/tests/preconditioners/fdp/fdp_test.i block=Preconditioning
+
+A more advanced check, and part of the process to fix the Jacobian, is to use the
+[Jacobian analysis utilities](development/analyze_jacobian.md). These will point out inaccurate or missing terms
+in the Jacobian. They can diagnose missing variable coupling terms for example. Once the errors have been identified,
```

Comment:
```suggestion
in the Jacobian. They can diagnose missing variable coupling terms for example. Jacobian deficiencies for small problems may also be investigated from the command line using the PETSc options `-snes_test_jacobian -snes_test_jacobian_view`. Combined with [DOFMap.md] output, this can be similarly used to diagnose problematic areas of the Jacobian. Once the errors have been identified,
```